### ISSUE TITLE
Feature Request: Provide a way to disable all system popups

### ISSUE THREAD

**Issue by matejdro** [android/android-test#406]
We are currently using android emulator to run periodic espresso tests. However, tests can occasionally fail if Android suddenly decides to open popup over our app, which makes all espresso matchers fail since our app is not the focused window anymore.

Example popups:

- System UI (or any preinstalled system app) has stopped / is not responding
- Viewing full screen - To exit swipe down

Maybe this could be an option in developer options or somehow triggered/enabled by espresso or UiAutmator.

**Comment by TWiStErRob** [comment]
Test butler already does this https://github.com/linkedin/test-butler#what-does-it-do, but a first-party implementation would be really nice here.

**Comment by matejdro** [comment]
Yes, I'm aware of test butler. Unfortunately, most of our apps rely on play services, and test butler cannot be used on Google API images, so we cannot use it.

**Comment by matejdro** [comment]
Just a heads up, there appears to be an [easy way to disable fullscreen popups](https://stackoverflow.com/questions/44357872/immersivemodeconfirmation-with-espresso-2-2):

> adb shell settings put secure immersive_mode_confirmations confirmed

**Comment by github-maxime-liege** [comment]
Any insights regarding this issue ?

I'm still having it on CI and i can't `adb` as i don't have the access to the device and need the google play services so i can't use TestButler
